evap/staff/forms.py

```diff
@@ -197,27 +197,41 @@ def save(self, *args, **kwargs):
 
 class CourseEmailForm(forms.Form, BootstrapMixin):
     recipients = forms.MultipleChoiceField(widget=forms.CheckboxSelectMultiple(), choices=EmailTemplate.EMAIL_RECIPIENTS, label=_("Send email to"))
-    subject = forms.CharField(label=_("Subject"))
-    body = forms.CharField(widget=forms.Textarea(), label=_("Message"))
+    subject = forms.CharField(label=_("Subject"), required=False)
+    body = forms.CharField(widget=forms.Textarea(), label=_("Message"), required=False)
 
     def __init__(self, *args, **kwargs):
         self.instance = kwargs.pop('instance')
         self.template = EmailTemplate()
+        self.export = False
         super().__init__(*args, **kwargs)
 
     def clean(self):
         self.recipient_groups = self.cleaned_data.get('recipients')
 
-        if not self.recipient_groups:
-            raise forms.ValidationError(_("No recipient selected. Choose at least one group of recipients."))
+        if self.export:
+            if not self.recipient_groups:
+                raise forms.ValidationError(_("No recipient selected. Choose at least one group of recipients."))
+            return self.cleaned_data
+
+        for field_name in ['recipients', 'subject', 'body']:
+            data = self.cleaned_data.get(field_name)
+            if not data:
+                self.add_error(field_name, _("Please fill out all required fields and select at least one recipient group."))
```

Comment:
i'm sorry, another suggestion. could you try handing over the export parameter into the constructor? and then pop it just like for the instance parameter. and then you could set the required parameter on the fields in the constructor depending on the export variable, and then remove those four lines here (and consequently show the `No recipient selected` always and not just `if export`.